crates/core/src/transcript/proof.rs

```diff
@@ -189,34 +240,34 @@ impl<'a> TranscriptProofBuilder<'a> {
         plaintext_hashes: &'a Index<PlaintextHashSecret>,
     ) -> Self {
         Self {
-            default_kind: TranscriptCommitmentKind::Encoding,
+            commitment_kinds: DEFAULT_COMMITMENT_KINDS.to_vec(),
             transcript,
             encoding_tree,
             plaintext_hashes,
-            encoding_query_idx: QueryIdx::new(),
-            hash_query_idx: QueryIdx::new(),
+            committed_idx: CommittedIdx::new(encoding_tree, plaintext_hashes),
+            query_idx: QueryIdx::new(),
         }
     }
 
-    /// Sets the default kind of commitment to open when revealing ranges.
-    pub fn default_kind(&mut self, kind: TranscriptCommitmentKind) -> &mut Self {
-        self.default_kind = kind;
+    /// Sets the commitment kinds in order of preference for building transcript
+    /// proofs, i.e. the first one is the most preferred.
+    pub fn commitment_kinds(&mut self, kinds: &[TranscriptCommitmentKind]) -> &mut Self {
+        if !kinds.is_empty() {
```

Comment:
Let's remove any duplicates which may user pass in `kinds` by mistake.

Reply:
good spot! fixed